### ISSUE TITLE
Sparse field sets return empty arrays for null attributes (shouldn't it be empty hashes)?

### ISSUE THREAD

**Issue by mmucklo** [woohoolabs/yin#5]
Back to the articles, comments, authors example:

What I'd like to do, for example, is to retrieve either all the body of comments on an article (but no information/attributes for the article), or all the first-names of authors on an article (again returning no attributes for the article).

According to this discussion:
http://discuss.jsonapi.org/t/clarification-on-the-sparse-fieldset-response/221/2

It seems that the proper way to do this is something like this:

```
GET /articles/1?include=comments&fields[articles]=&fields[comments]=body
```

It should return something like this (as copied/interpreted form the discussion linked above):

``` json
{
  "data": {
     "type": "articles",
     "id": "1",
   },
  "included": [{
    "type": "comments",
    "id": "1",
    "attributes": { "body": "yadayada"},
  }]
}
```

~~However with yin, it seems it's not removing the articles attributes, relationships etc but returning them...~~
## Update 1/19/2016:

It's been determined this was due to user error.  Yin does seem to properly null out attributes (cool!).  There's a small other issue which I've highlighted below in a new comment (updating the title of this issue to reflect it)

**Comment by kocsismate** [comment]
I am not sure about this issue: I tested the sparse fieldset and inclusion functionality with my book example and with another project of mine, but they both worked well. I couldn't make Yin fail with any queries I came up with and it even worked with 2 level deep relationships.

Could you show me an actual query and its response? If it is not possible, could you double check your settings?

**Comment by mmucklo** [comment]
Let me try to put together a test case..

**Comment by mmucklo** [comment]
Ok, I'll chalk this one up to user error.

However I think I may have discovered a very minor issue.

The json returned for null attributes is a null array, but in the link I referenced it seems to show null hashes:

http://discuss.jsonapi.org/t/clarification-on-the-sparse-fieldset-response/221/2

Test Url:

```
curl --url "http://localhost:4000/examples/index.php?example=book&id=1&include=authors,publisher,publisher.representative&fields\[book\]="
```

Response:

``` json
{"jsonApi":{"version":"1.0"},"links":{"self":"\/books\/1"},"data":{"type":"book","id":"1","links":{"self":"\/books\/1"},"attributes":[],"relationships":[]}, ....

```

**Comment by kocsismate** [comment]
I am glad that it was only user error. :)

The other issue stands because Yin uses arrays internally to store the document and `json_encode` converts empty arrays to JSON arrays. Tell me if you know a workaround for this phenomenon. :)

However, according to the discussion you linked, a possible easy fix would be to omit these keys altogether when empty. I'll implement it!

**Comment by kocsismate** [comment]
I am closing the issue as the problem should be fixed in v0.10.1!  Feel free to reopen the bug if you think otherwise.
